### DISCUSSION TITLE
format of Grafana agent remote_write request

### DISCUSSION THREAD

**Discussion by ruwalig** [grafana-cold-storage/agent#810] [Q&A]
Hi all, I am new to observability and was trying my hands with Grafana agent. I have a query on the remote_write request that Grafana agent uses for pushing the metrics. What format does the agent uses for remote_write request(json....or something else) ?

**Comment by rfratto** [comment]
Thanks for your question! We rely on Prometheus' code for sending metrics. It has used snappy-compressed protobuf for sending metrics for a while. The remote_write protocol has recently been [formalized](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit#heading=h.n0d0vphea3fe), feel free to read that if you want more detail.

**Reply by ruwalig** [reply]
thanks a lot for answering my query.
